## TensorFlow Keras

In [65]:
import numpy as np
import pandas as pd
import tensorflow as tf

The CSV files contains ---
- other columns represents various features on which the result depends
- smoking column which indicates Y/N for smokers and non-smokers


In [66]:
df = pd.read_csv("/content/smoking.csv")

map = {'Y':1,'N':0}
gender = {'M':1, 'F':0}

df['gender'] = df['gender'].map(gender)
df['oral'] = df['oral'].map(map)
df['tartar'] = df['tartar'].map(map)

df.head(5)

ID  gender  age  height(cm)  weight(kg)  ...   Gtp  oral  dental caries  tartar  smoking
0   0       0   40         155          60  ...  27.0     1              0       1        0
1   1       0   40         160          60  ...  18.0     1              0       1        0
2   2       1   55         170          60  ...  22.0     1              0       0        1
3   3       1   40         165          70  ...  18.0     1              0       1        0
4   4       0   40         155          60  ...  22.0     1              0       0        0

[5 rows x 27 columns]

In [67]:
X = np.array(df.loc[:, ~df.columns.isin(['ID', 'smoking'])])
y = np.array(df[["smoking"]])

m=X.shape[0]

print(X.shape)

X_train = X[:int(m*0.95)]
y_train = y[:int(m*0.95)]

print(X_train.shape)

X_test = X[:m-int(m*0.95)]
y_test = y[:m-int(m*0.95)]

print(X_test.shape)


(55692, 25)
(52907, 25)
(2785, 25)


In [71]:
model = tf.keras.models.Sequential([
    tf.keras.Input(shape=(25,)),
    tf.keras.layers.Dense(units=10, activation="relu", name="layer1"),
    tf.keras.layers.Dense(units=4, activation="relu", name="layer2"),
    tf.keras.layers.Dense(units=1, activation="sigmoid", name="layer3"),
])

model.compile(loss=tf.keras.losses.BinaryCrossentropy(), metrics=['accuracy'])
model.fit(X_train, y_train, epochs=20)

Epoch 1/20
1654/1654 [==============================] - 5s 3ms/step - loss: 0.6436 - accuracy: 0.6327
Epoch 2/20
1654/1654 [==============================] - 3s 2ms/step - loss: 0.6027 - accuracy: 0.6394
Epoch 3/20
1654/1654 [==============================] - 3s 2ms/step - loss: 0.5799 - accuracy: 0.6721
Epoch 4/20
1654/1654 [==============================] - 3s 2ms/step - loss: 0.5660 - accuracy: 0.6837
Epoch 5/20
1654/1654 [==============================] - 4s 3ms/step - loss: 0.5560 - accuracy: 0.6933
Epoch 6/20
1654/1654 [==============================] - 3s 2ms/step - loss: 0.5473 - accuracy: 0.6982
Epoch 7/20
1654/1654 [==============================] - 3s 2ms/step - loss: 0.5409 - accuracy: 0.7038
Epoch 8/20
1654/1654 [==============================] - 4s 2ms/step - loss: 0.5336 - accuracy: 0.7067
Epoch 9/20
1654/1654 [==============================] - 4s 2ms/step - loss: 0.5239 - accuracy: 0.7137
Epoch 10/20
1654/1654 [==============================] - 3s 2ms/step - loss: 0.512

In [72]:
loss, accuracy = model.evaluate(X_test, y_test)

print(f"Loss - {loss}")
print(f"Accuracy - {accuracy}")

88/88 [==============================] - 0s 2ms/step - loss: 0.4910 - accuracy: 0.7436
Loss - 0.4910430610179901
Accuracy - 0.743626594543457


epoch = 10, layer=[10,4,1]

- relu, relu, sigmoid --- 0.7325
- relu, sigmoid, sigmoid --- 0.728
- sigmoid, sigmoid, sigmoid --- 0.734

epoch = 20, layer=[10,4,1]

- relu, relu, sigmoid --- 0.63
- relu, sigmoid, sigmoid --- 0.741
- sigmoid, sigmoid, sigmoid --- 0.744